database/database.go

```diff
@@ -93,20 +90,18 @@ func (db *DB) Disconnect(ctx context.Context) error {
 	return db.staticDB.Client().Disconnect(ctx)
 }
 
-// UserByEmail returns the user with the given email or nil.
-func (db *DB) UserByEmail(ctx context.Context, email Email) (*User, error) {
-	if !email.Validate() {
-		return nil, ErrInvalidEmail
-	}
-	users, err := db.managedUsersByField(ctx, "email", string(email))
+// UserBySub returns the user with the given sub or nil. The sub is the Kratos
+// id of that user.
+func (db *DB) UserBySub(ctx context.Context, sub string) (*User, error) {
+	users, err := db.managedUsersByField(ctx, "sub", sub)
 	if err != nil {
 		return nil, err
 	}
-	// Emails must be unique. If we hit this then we have a serious
+	// Subs must be unique. If we hit this then we have a serious
 	// programmer error which endangers customer's data and finances.
 	// We should error out in order to prevent exposing the wrong user data.
 	if len(users) > 1 {
-		build.Critical(fmt.Sprintf("More than one user found for email '%s'!", email))
+		build.Critical(fmt.Sprintf("More than one user found for sub '%s'!", sub))
 		// The error message is intentionally cryptic.
 		return nil, ErrGeneralInternalFailure
```

Comment:
This is fine but I like `Failed Dependency` a lot for this type of stuff personally. It makes it look like it's bricked, but not entirely our fault perhaps :D less shameful than a 500 error

Reply:
I guess it won't be the fault of a dependency in this case - it seems like something we messed up, not a DB that's failing?

Reply:
Yeah, we would be the dependency 😁  returning 500 error is just not done in my book. But perhaps that's just something that someone once told me that I now think is not done. I was once told by previous mentors that returning a 500 on purpose when designing an API is something you generally don't do.

Again, feel free to resolve.

Reply:
In my experience a 500 denotes any error that's caused by the service's own code.